code/datums/viewport.dm

```diff
@@ -0,0 +1,72 @@
+/client/ // Holder Variables
+	var/last_view_x_dim = 7
+	var/last_view_y_dim = 7
+
+// Used in OnResize() - Holds the clients prefs.
+GLOBAL_VAR_INIT(lock_client_view_x, null)
+GLOBAL_VAR_INIT(lock_client_view_y, null)
+GLOBAL_LIST_INIT(valid_icon_sizes, list(32, 48, 64, 96, 128))
+
+// We hold this so we don't hit prefs often.
+/datum/preferences/var/icon_size = 64
+
+/client/verb/SetWindowIconSize(val as num|text)
+	set hidden = 1
```

Comment:
shouldn't this be a game preference rather then a verb?

Reply:
It's a verb you can type in your command bar to pick odd/non-listed scales for people with weirder screens

Reply:
Why not make it a game preference? it's weird to do it as a verb when we have a system for this sort of thing

Reply:
Well... I find preferences a bit janky for this sort of thing, but, I am indeed thinking I should move it there and add a hotkey of some sort to increment the size by 2 ++ or --.

Reply:
Aight this is all game pref based now

Though the skin portion remains at the moment